base/intfuncs.jl

```diff
@@ -1075,3 +1075,31 @@ function binomial(n::T, k::T) where T<:Integer
     end
     copysign(x, sgn)
 end
+
+"""
+    iroot(n::Integer, r::Integer)
+
+Computes floor(n^(1/r)) precisely.
+# Examples
+```jldoctest
+julia> iroot(100, 2)
+10
+
+julia> iroot(99,2)
+9
+```
+"""
+iroot(x::Integer, n::Integer) = iroot(promote(x, n)...)
```

Comment:
The meaning of `n` is different in the method signature (and error messages) and the doc-string. I think this would be nice to have consistent across methods (I got confused, at least).

I.e. could the method signatures be `iroot(n, r)` as they are in the doc-string, instead of the current `iroot(x, n)`?

Reply:
I switched the docs to `x,n` since that is the typical notation (ie the nth root).